garrysmod/lua/includes/extensions/util.lua

```diff
@@ -347,3 +347,64 @@ function util.RemovePData( steamid, name )
 	sql.Query( "DELETE FROM playerpdata WHERE infoid = "..SQLStr(name) )
 	
 end
+
+--[[---------------------------------------------------------
+   Name: ClipTraceToPlayers
+   Desc: Traces across players and returns a modified trace
+-----------------------------------------------------------]]
+local function DistanceToTrace( pos, traceStart, traceEnd, along, pointOnTrace )
+	local to = pos - traceStart
+	local dir = traceEnd - traceStart
+	//Dunno if NormalizeInPlace translates to GetNormalized or Normalize

```

Comment:
The benefit of in-place mutation is that you can skip the overhead of creation of a new object.